samples/csharp_dotnetcore/49.qnamaker-all-features/Dialog/QnAMakerBaseDialog.cs

```diff
@@ -1,5 +1,6 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
```

Comment:
What if we update the methods to be synchronous? Like this:
```CSharp
protected IQnAMakerClient GetQnAMakerClient()
{
    return _services?.QnAMakerService;
}
```

Reply:
It wasn't used. Anyway, I think it'd be better to keep the override of the method and add the `Task.FromResult()` as I suggested for the teams sample.